### PR TITLE
Disable rich workspace rendering when view is not visible

### PR DESCRIPTION
**Pull request by juliushaertl**
- Do not render the workspace if file list is hidden (fixes https://github.com/nextcloud/text/issues/2807)
- Add cypress tests for switching views with rich workspaces (fixes https://github.com/nextcloud/text/issues/1905)

As the file list is just hidden by the files app the vue component would still be active when switching to recent files for example.

### CODE REVIEW

**Comment by juliushaertl** [comment]
The event itself is available since 24 by the backports of https://github.com/nextcloud/server/pull/34100

**Review by mejo- — COMMENTED** [review]
The fix works well when tested locally. Cypress tests seem to need some love though.

**Comment by juliushaertl** [comment]
Strange, it was passing fine locally, but I'll dive into that.

**Review by max-nextcloud — CHANGES_REQUESTED** [review]
Code looks good.

Requesting some minor changes that will make retries use consistent folder names.

File: cypress/e2e/workspace.spec.js [hunk 1]
```diff
@@ -43,6 +43,32 @@ describe('Workspace', function() {
...
+	it('Hides the workspace when switching to another folder', function() {
+		cy.uploadFile('test.md', 'text/markdown', `${Cypress.currentTest.title}/README.md`)
+		cy.createFolder(`${Cypress.currentTest.title}/subdirectory`)
+		cy.reload()
+		cy.get('.files-fileList').should('contain', 'README.md')
+		cy.get('#rich-workspace .ProseMirror')
+			.should('contain', 'Hello world')
+		cy.openFolder('subdirectory')
+		cy.get('#rich-workspace')
+			.get('.ProseMirror')
```
Comment (max-nextcloud) [inline cypress/e2e/workspace.spec.js:55]: The first get already fails. So i guess you want this:
```suggestion
		cy.get('#rich-workspace .ProseMirror')
```

File: cypress/e2e/workspace.spec.js [hunk 2]
```diff
@@ -43,6 +43,32 @@ describe('Workspace', function() {
...
+		cy.uploadFile('test.md', 'text/markdown', `${Cypress.currentTest.title}/README.md`)
+		cy.reload()
+		cy.get('.files-fileList').should('contain', 'README.md')
+		cy.get('#rich-workspace .ProseMirror')
+			.should('contain', 'Hello world')
+		cy.get('.nav-recent')
+			.click()
+		cy.get('#rich-workspace')
+			.get('.ProseMirror')
+			.should('not.exist')
```
Comment (max-nextcloud) [inline cypress/e2e/workspace.spec.js:69]: `#rich-workspace` does not exist at all in this scenario. So the first get fails.
```suggestion
		cy.get('#rich-workspace .ProseMirror')
			.should('not.exist')
```

**Review by max-nextcloud — APPROVED** [review]

**Review by mejo- — APPROVED** [review]
Thanks for fixing the tests @max-nextcloud.

**Comment by mejo-** [comment]
/backport 581233665bb7516c2da5cd3bd74f030c08ff10f2,f85a3cd5c996dbfe8d2230859d67ae5c17847242,9fed8ea72469d6d25c6723689aa5c2b634ea1a52 to stable25

**Comment by juliushaertl** [comment]
Failures/flaky tests are known, merging

**Comment by mejo-** [comment]
/backport 581233665bb7516c2da5cd3bd74f030c08ff10f2,f85a3cd5c996dbfe8d2230859d67ae5c17847242,9fed8ea72469d6d25c6723689aa5c2b634ea1a52 to stable24

**Pull request merged**